lua/llm/providers/llamacpp_server.lua

```diff
@@ -0,0 +1,115 @@
+local system = require('llm.util.system')
+
+local curl = require "llm.curl"
+local util = require "llm.util"
+local provider_util = require "llm.providers.util"
+
+local M = {}
+
+---@param handlers StreamHandlers
+---@param params? any Additional params for request
+---@param options? { model?: string }
+function M.request_completion(handlers, params, options)
+  local model = (options or {}).model or "bigscience/bloom"
+  -- vim.print(params)
+
+  -- TODO handle non-streaming calls
+  return curl.stream({
+    -- url = 'https://api-inference.huggingface.co/models/', --.. model,
+    url = "http://127.0.0.1:8080/completion",
+    method = "POST",
+    body = vim.tbl_extend("force", { stream = true }, params),
+    headers = {
+      -- Authorization = 'Bearer ' .. util.env_memo('HUGGINGFACE_API_KEY'),
+      ["Content-Type"] = "application/json",
+      -- ['data'] = '{"prompt": "Building a website can be done in 10 simple steps:","n_predict": 128}',
+    },
+  }, function(raw)
+
+    provider_util.iter_sse_items(raw, function(item)
+      local data = util.json.decode(item)
+
+      if data == nil then
+        handlers.on_error(item, "json parse error")
+        return
+      end
+
+      if data.generation_settings ~= nil then -- last message
+        handlers.on_finish('', "stop")
```

Comment:
Can just be a naked call now that it's handled in `provider.lua`
```suggestion
        handlers.on_finish()
```

Reply:
done